rules-tests/TypeDeclaration/Rector/Class_/TypedPropertyFromCreateMockAssignRector/Fixture/skip_non_setup_method.php.inc

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\Class_\TypedPropertyFromCreateMockAssignRector\Fixture;
+
+use PHPUnit\Framework\TestCase;
+use Rector\Tests\TypeDeclaration\Rector\Class_\TypedPropertyFromCreateMockAssignRector\Source\SomeMockedClass;
+
+class SkipNonSetupMethod extends TestCase
+{
+    public $someMock;
+
+    protected function factory(): void
+    {
+        $this->someMock = $this->createMock(SomeMockedClass::class);
```

Comment:
@TomasVotruba do you think `$someMock` need to be `?\PHPUnit\Framework\MockObject\MockObject` as well when there is no `setUp()` method?

In that case, I think Property guard is needed to verify only `private` property, or allow `protected` property only on final class, by this service `Rector\Php74\Guard\PropertyTypeChangeGuard`

in case of protected property, then we can use the `FinalizeTestCaseClassRector` to allow `protected` property changed after class finalized.

Reply:
> $someMock need to be ?\PHPUnit\Framework\MockObject\MockObject as well when there is no setUp() method?

Yes, as it would crash if accessed without nullable.

Reply:
> In that case, I think Property guard is needed to verify only private property, or allow protected property only on final class, by this service Rector\Php74\Guard\PropertyTypeChangeGuard

Not really, scope of this rule is quite limited. It's unlikely tests and mocks will get overriden in child tests.

Reply:
This fixture should behave like following:

```diff
-public $someMock;
+public ?MockObject $someMock;
```

Reply:
sure, updated and add `null` default value as not assigned early on `setUp()` https://github.com/rectorphp/rector-src/pull/6180/commits/1da4c41e129906e62f62bf3efc13eab2ac9400fe